### PR TITLE
Jumpbox kubeconfig

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**:  installs kubectl + kubeconfig on a private cluster jumpbox

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: parts/k8s/kubernetesjumpboxcustomdata.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#cloud-config
+
+write_files:
+
+- path: "/home/{{WrapAsVariable "jumpboxUsername"}}/.kube/config"
```
Comment (jackfrancis) [inline parts/k8s/kubernetesjumpboxcustomdata.yml]: Is there a possible race condition between cloud-init and user `<jumpBoxUsername>` creation on the host OS?

Comment (CecileRobertMichon) [reply parts/k8s/kubernetesjumpboxcustomdata.yml]: I don't think so jumpboxUsername is a variable just like we use variables in master custom data, it stays as a variable up until deployment and jumpboxUsername is passed in from the apimodel

File: parts/k8s/kubernetesjumpboxcustomdata.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+write_files:
+
+- path: "/home/{{WrapAsVariable "jumpboxUsername"}}/.kube/config"
+  permissions: "0744"
+  encoding: "gzip"
+  owner: "root"
+  content: |
+    {{GetKubeConfig}}
+
+runcmd:
```
Comment (jackfrancis) [inline parts/k8s/kubernetesjumpboxcustomdata.yml]: Add this beauty:

`retrycmd_if_failure() { retries=$1; wait=$2; shift && shift; for i in $(seq 1 $retries); do ${@}; [ $? -eq 0  ] && break || sleep $wait; done; echo Executed \"$@\" $i times; }`

And then change the curl statement accordingly, e.g.:

`retrycmd_if_failure 10 5 curl -LO https://storage.googleapis.com/kubernetes-release/release/v{{.OrchestratorProfile.OrchestratorVersion}}/bin/linux/amd64/kubectl`

File: parts/k8s/kubernetesjumpboxcustomdata.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#cloud-config
+
+write_files:
+
+- path: "/home/{{WrapAsVariable "jumpboxUsername"}}/.kube/config"
+  permissions: "0744"
```
Comment (jackfrancis) [inline parts/k8s/kubernetesjumpboxcustomdata.yml]: I think this should be `"0644"`

File: parts/k8s/kubernetesjumpboxcustomdata.yml (hunk 1 above)
Comment (jackfrancis) [reply parts/k8s/kubernetesjumpboxcustomdata.yml]: Sorry if I wasn't clear: `/home/<jumpboxUsername>/` won't exist until the user `<jumpboxUsername>` is created. What process creates that user, and does it happen before cloud-init?

Comment (CecileRobertMichon) [reply parts/k8s/kubernetesjumpboxcustomdata.yml]: I checked and it's not the case

**Pull request merged**